__tests__/cron/updateCurrentStreak.ts

```diff
@@ -81,30 +80,156 @@ describe('updateCurrentStreak cron', () => {
       .getRepository(UserStreak)
       .update({ userId: '1' }, { lastViewAt: new Date('2024-06-21') });
     // This is a Monday
-    jest
-      .useFakeTimers({
-        doNotFake: [
-          'hrtime',
-          'nextTick',
-          'performance',
-          'queueMicrotask',
-          'requestAnimationFrame',
-          'cancelAnimationFrame',
-          'requestIdleCallback',
-          'cancelIdleCallback',
-          'setImmediate',
-          'clearImmediate',
-          'setInterval',
-          'clearInterval',
-          'setTimeout',
-          'clearTimeout',
-        ],
-      })
-      .setSystemTime(new Date('2024-06-24'));
+    jest.useFakeTimers({ doNotFake }).setSystemTime(new Date('2024-06-24'));
     await expectSuccessfulCron(cron);
     const streak = await con
       .getRepository(UserStreak)
       .findOneBy({ userId: '1' });
     expect(streak.currentStreak).toBe(1);
   });
+
+  describe('incorporating streaks recovery', () => {
+    beforeEach(() => {
+      nock('http://localhost:5000').post('/e').reply(204);
+    });
+
+    it('should not reset streak if user has recovered date of today', async () => {
+      /*
+      Day today is Wednesday
+      1. Last read was Monday
+      2. User did not read anything on Tuesday
+      3. Wednesday arrived and resets the streak to 0
+      4. User recovers the streak on Wednesday
+      5. When the cron job runs, streak should stay at 1
```

Comment:
Can you add a test here where this happens, but user also reads on Wednesday so now streak should be 2?

Maybe we should actually test both these:
- User reads first -> then resets
- User resets first -> then reads

Reply:
Sure, let's add those ones.

Reply:
> User resets first -> then reads

Added a test now but within the `newView` test suite to make it more realistic in terms of adding a read to update the streak.

> User reads first -> then resets

We should have at least 3 tests on this.